### PR TITLE
✨ Add serviceAccountName to DeploymentSpec

### PR DESCRIPTION
**Pull request by steve-fraser**
**What this PR does / why we need it**:

Today, it's difficult to override the service account with an IRSA configured value. I would like the ability override the serviceAccountName for the deployment

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by Fedosin** [comment]
Fix for the linter issue: https://github.com/kubernetes-sigs/cluster-api-operator/pull/176

**Review by furkatgofurov7 — COMMENTED** [review]
/lgtm

**Comment by furkatgofurov7** [comment]
@steve-fraser thanks for the PR. Can you please squash your commits, otherwise LGTM.
/lgtm cancel
/hold lgtm pending squash

**Review by alexander-demicev — COMMENTED** [review]
Thanks for the PR, we can merge it after commits are squashed.

**Review by furkatgofurov7 — COMMENTED** [review]
Thanks @steve-fraser!

/lgtm
/cc @alexander-demicev for approval

**Review by alexander-demicev — APPROVED** [review]
/approve

**Pull request merged**
